code/modules/mob/mob.dm

```diff
@@ -279,22 +279,25 @@
 	return
 
 /mob/proc/reset_view(atom/A)
-	if (client)
-		client.pixel_x = initial(client.pixel_x)
-		client.pixel_y = initial(client.pixel_y)
-		A = A ? A : eyeobj
-		if (istype(A, /atom/movable))
-			client.perspective = EYE_PERSPECTIVE
-			client.eye = A
-		else
-			if (isturf(loc))
-				client.eye = client.mob
-				client.perspective = MOB_PERSPECTIVE
-			else
-				client.perspective = EYE_PERSPECTIVE
-				client.eye = loc
-	return
-
+	set waitfor = 0
+	while(shakecamera && client && !QDELETED(src))
+		sleep(1)
```

Comment:
Probably should be `stoplag(1)` instead of sleep.

Reply:
On second thought, these're all nullchecks - no need.

Reply:
As discussed on Discord, would prefer to leave it as `sleep(1)` for clarity.